### PR TITLE
BS-558: Added Underpayments feature flag

### PR DESCRIPTION
**Pull request by tgeary90**

### CODE REVIEW

**Comment by tgeary90** [comment]
`2022-02-03 12:14:27,777 level=[INFO] logger=[uk.gov.hmrc.breathingspaceifproxy.controller.UnderpaymentsController] thread=[application-akka.actor.default-dispatcher-6] rid=[] user=[] message=[Underpayments feature enabled : true] `

File: test/uk/gov/hmrc/breathingspaceifproxy/controller/UnderpaymentsControllerSpec.scala [hunk 1]
```diff
@@ -154,5 +159,47 @@ class UnderpaymentsControllerSpec extends AnyWordSpec with BaseSpec with Mockito
...
+
+      val response = flaggedOffProxy.get(validNino, validPeriodId)(fakeGetRequest)
+
+      status(response) shouldBe NOT_IMPLEMENTED
+    }
+
+    "Underpayments feature switch should return 200 when flag set to true" in {
+      val flaggedOffConfig: Configuration = Configuration(ConfigFactory.load("underpaymentsOn.conf"))
+      val servicesConfig = inject[ServicesConfig]
+      val flaggedOnAppConfig = new AppConfig(flaggedOffConfig, servicesConfig)
```
Comment (macgrewal) [inline test/uk/gov/hmrc/breathingspaceifproxy/controller/UnderpaymentsControllerSpec.scala]:
```suggestion
      val featureConfig = Configuration.from(Map("feature.flag.underpayments.enabled" -> true))
      val flaggedOnConfig = featureConfig.withFallback(inject[Configuration])
      val servicesConfig = inject[ServicesConfig]
      val flaggedOnAppConfig = new AppConfig(flaggedOnConfig, servicesConfig)
```

File: test/uk/gov/hmrc/breathingspaceifproxy/controller/UnderpaymentsControllerSpec.scala [hunk 2]
```diff
@@ -154,5 +159,47 @@ class UnderpaymentsControllerSpec extends AnyWordSpec with BaseSpec with Mockito
       errorList.last.code shouldBe INVALID_NINO.entryName
       assert(errorList.last.message.startsWith(INVALID_NINO.message))
     }
+
+    "Underpayments feature switch should return 501 when flag set to false" in {
+      val flaggedOffConfig: Configuration = Configuration(ConfigFactory.load("underpaymentsOff.conf"))
+      val servicesConfig = inject[ServicesConfig]
+      val flaggedOffAppConfig = new AppConfig(flaggedOffConfig, servicesConfig)
```
Comment (macgrewal) [inline test/uk/gov/hmrc/breathingspaceifproxy/controller/UnderpaymentsControllerSpec.scala]:
```suggestion
      val featureConfig = Configuration.from(Map("feature.flag.underpayments.enabled" -> false))
      val flaggedOffConfig = featureConfig.withFallback(inject[Configuration])
      val servicesConfig = inject[ServicesConfig]
      val flaggedOffAppConfig = new AppConfig(flaggedOffConfig, servicesConfig)
```

**Review by macgrewal — CHANGES_REQUESTED** [review]
The `.conf` files in the test resources can be removed and config can be overridden in the tests

File: test/uk/gov/hmrc/breathingspaceifproxy/controller/UnderpaymentsControllerSpec.scala (hunk 1 above)
Comment (tgeary90) [reply test/uk/gov/hmrc/breathingspaceifproxy/controller/UnderpaymentsControllerSpec.scala]: done

**Review by macgrewal — APPROVED** [review]

**Pull request merged**